src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,560 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long, too-many-locals, too-many-statements, too-many-branches, too-few-public-methods
+
+from azure.appconfiguration import AzureAppConfigurationClient, ConfigurationSetting
+from azure.appconfiguration._generated.models import Error as AppConfigError
+from azure.core.rest import HttpRequest
+from azure.core.paging import ItemPaged
+from azure.core.utils import case_insensitive_dict
+from azure.core.exceptions import ClientAuthenticationError, ResourceExistsError, ResourceNotFoundError, HttpResponseError, ResourceModifiedError, map_error
+from msrest import Serializer
+import json
+
+from ._constants import SnapshotConstants
+from ._snapshotmodels import Snapshot, SnapshotListResult, OperationStatusResponse
+from ._snapshot_exceptions import BadSnapshotRequestException
+
+
+class ProvisioningStatus:
+    RUNNING = "Running"
+    SUCCEEDED = "Succeeded"
+    FAILED = "Failed"
+
+
+class RequestMethod:
+    GET = "GET"
+    PUT = "PUT"
+    PATCH = "PATCH"
+    DELETE = "DELETE"
+
+
+_ERROR_MAP = {
+    400: BadSnapshotRequestException,
+    401: ClientAuthenticationError,
+    404: ResourceNotFoundError,
+    409: ResourceExistsError,
+    412: ResourceModifiedError
+}
+
+
+def _build_request(
+    template_url_default,
+    method,
+    *,
+    path_arguments=None,
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    api_version = kwargs.pop("api_version", SnapshotConstants.API_VERSION)
```

Comment:
In addition, if you have new features which are needed to be developed in the future *(especially for atomic commands)*, we strongly recommend that you try our advanced code generation tools [CodeGen - AAZ](https://azure.github.io/aaz-dev-tools/)

This tool is easy to use and can bring you the following benefits:
- It can improve your development efficiency and reduce manual effort.
- It can ensure the style and consistency of code, so it also reduces the burden of PR review.
- It can improve code quality and reduce the occurrence of bugs.
......

If you are interested in it and would like to learn more, please contact our CodeGen expert @kairu-ms